TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc

```diff
@@ -535,8 +535,13 @@ namespace mu2e {
 
     void StrawDigisFromStrawGasSteps::fillClusterMap(StrawPhysics const& strawphys,
         StrawElectronics const& strawele,
-        TrackerStatus const& trackerStatus,
         art::Event const& event, StrawClusterMap & hmap){
+// get status if needed
+      std::shared_ptr<const TrackerStatus> trackerStatus;
+      if(_usestatus) {
+        ProditionsHandle<TrackerStatus> _trackerStatus_h;
```

Comment:
@rlcee I have a question about the recommended use patterns of proditions .   Maybe I am out of date but I thought that the plan was to get the ProditionsHandle once at the start of the job and hold it as a data member.  It does not occupy much memory and but getting a new handle every event is a little bit of work.   If _usedb is false, you will never call getPtr(event.id()).  While it's not really important in this module I think we want to establish a recommended pattern and show it to others as a module.

Reply:
Yes, the ProditionsHandle should be created once as an art module class member.  On construction, it will get a pointer from the service, but nothing substantial happens.  calling get() or getPtr() triggers fromFcl or fromDb, depending on that setting, so if you are not going to use the result, you probably want to not call that.  Recreating the handle in the event will cause it to get that service pointer again, then also search for the cached content that you are losing the pointer to when you deleted the handle last event.  So if you are using the content, then deleting the handle frequently is a little cost.

Reply:
thanks for looking after this.